src/utils/auth_utils.js

```diff
@@ -62,38 +91,23 @@ const logInWithEmailAndPassword = async (email, password, redirectPath, navigate
   await signInWithEmailAndPassword(auth, email, password);
   navigate(redirectPath);
   cookies.set(cookieKeys.ACCESS_TOKEN, auth.currentUser.accessToken, cookieConfig);
+  addRoleToCookies(cookies);
 };
 
 const createUserInFirebase = async (email, password) => {
   const user = await createUserWithEmailAndPassword(auth, email, password);
   sendEmailVerification(user.user);
```

Comment:
please move `sendEmailVerification` to after the user was successfully added to our database. In the current workflow, a user can be added to firebase and an email verification will be sent. However, in the chance where there is an error when trying to add the user to our database, that user will then be deleted from firebase. This will cause an email verification to be sent even though the user is deleted.